[MB-4745] Add payment requests needing approval count to Payment requests tab

hannaliebl opened the pull request:
## Description

~~Draft PR for now to make sure this all seems good. I would like to add tests, but otherwise am looking to see if this approach is correct.~~ EDIT: I have added some tests for what I added, and moved the existing enzyme tests for the `MovePaymentRequests` component to react-testing-library.

My own Todo:
* [X] Add tests

## Reviewer Notes

Just ensure I'm not missing any edgecases and that the logic of displaying the indicator makes sense.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

Login as tio_role@office.mil role

## Code Review Verification Steps

`MBRYPJ` move code has no indicator as payment request has been reviewed already
`TXKRK9` has one indicator, reviewing it removes indicator


* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4745) for this change

## Screenshots
![Screen Shot 2021-03-16 at 3 20 57 PM](https://user-images.githubusercontent.com/3903208/111367559-3db7a400-866b-11eb-8ddc-fd4d521dccab.png)

robot-mymove commented:
<!--
  0 failure: 
  1 warning:  This PR does not ...
  1 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to unit tests, even though it affects app code.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:link: <a href="https://dp3.atlassian.net/browse/MB-4745">MB-4745</a></td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 564c4e72ae7ea2c012fec6b69118452d17533035
</p>

duncan-truss commented:
Your approach looks great!

duncan-truss commented on src/pages/Office/MovePaymentRequests/MovePaymentRequests.jsx:
```diff
@@ -3,14 +3,15 @@ import { useParams } from 'react-router-dom';
 import { func } from 'prop-types';
 
 import txoStyles from '../TXOMoveInfo/TXOTab.module.scss';
+import { PAYMENT_REQUEST_STATUS } from '../../../shared/constants';
```
We have been trying to move constants out of the `shared/` folder. It looks like there are already some defined in `constants/paymentRequestStatus.js` but we need to add the missing `REVIEWED_AND_ALL_SERVICE_ITEMS_REJECTED` status to it.

jenniferlynparsons reviewed (commented):
Looks good barring the suggested constant change. 🎉

hannaliebl commented:
Ready for review again (I also moved it out of Draft status.) I updated this to use the constants not from /shared. I also added some tests, and rewrote the existing Enzyme tests for the `MovePaymentRequests` in react-testing-library.

duncan-truss commented on src/pages/Office/MovePaymentRequests/MovePaymentRequests.test.jsx:
```diff
@@ -83,46 +87,114 @@ const multiplePaymentRequests = {
+  describe('with multiple payment requests', () => {
+    beforeEach(() => {
+      useMovePaymentRequestsQueries.mockImplementation(() => multiplePaymentRequests);
+    });
```
Nice, I haven't used testing-library much and will need to pull this into my branch.

Do we need to set the mock implementation again between each it block?  Similar question about rerendering the component for each it block if the data and component aren't being manipulated?

hannaliebl replied:
Ah, ok, so I did some digging on the render thing, and I think React Testing Library automatically unmounts after every test: https://testing-library.com/docs/react-testing-library/setup/#skipping-auto-cleanup so I do think I need to render it in every test.

As for the before each, I ran into issues with tests passing if I didn't put it into a beforeEach block. There may be a more optimal way to do this, though.

duncan-truss replied:
👍 Thanks for looking into this, I learned something new.

duncan-truss reviewed (approved):
🎉 First PR! First PR! 💥

brooksr4 reviewed (approved):
Looks good
